src/client/voice/receiver/PacketHandler.js

```diff
@@ -78,6 +79,23 @@ class PacketHandler extends EventEmitter {
     const ssrc = buffer.readUInt32BE(8);
     const user = this.userFromSSRC(ssrc);
     if (!user) return;
+
+    if (this.speakingTimeouts.has(ssrc)) {
+      clearTimeout(this.speakingTimeouts.get(ssrc));
```

Comment:
This is an anti-pattern, plus it makes the code check the map's entries twice, you can do the following instead:

```javascript
const previousTimeout = this.speakingTimeouts.get(ssrc);
if (typeof previousTimeout === 'undefined') {
  this.connection.onSpeaking({ user_id: user.id, ssrc, speaking: 1 });
} else {
  clearTimeout(previousTimeout);
  this.speakingTimeouts.delete(ssrc);
}
```